crates/bevy_reflect/src/struct_trait.rs

```diff
@@ -73,6 +73,10 @@ pub trait Struct: Reflect {
 
     /// Clones the struct into a [`DynamicStruct`].
     fn clone_dynamic(&self) -> DynamicStruct;
+
+    /// Returns the structs [`StructInfo`].
+    /// Will return `None` if [`TypeInfo`] is not available.
+    fn get_represented_kind_info(&self) -> Option<&StructInfo>;
```

Comment:
We could make this a method on `Reflect` and then use an associated type to control the return value, right? Would that be better?

Reply:
Hmm, actually, that's a bad idea. Because then we can't do a `Box<dyn Reflect>` without specifying the associated type.